src/compute/src/logging/compute.rs

```diff
@@ -890,3 +901,74 @@ impl Drop for RetractImportFrontiers {
         self.log();
     }
 }
+
+/// Extension trait to attach `ComputeEvent::DataflowError` logging operators to collections and
+/// batch streams.
+pub(crate) trait LogDataflowErrors {
+    fn log_dataflow_errors(self, logger: Logger, export_id: GlobalId) -> Self;
+}
+
+impl<G, D> LogDataflowErrors for Collection<G, D, Diff>
+where
+    G: Scope,
+    D: Data,
+{
+    fn log_dataflow_errors(self, logger: Logger, export_id: GlobalId) -> Self {
+        self.inner
+            .unary(Pipeline, "LogDataflowErrors", |_cap, _info| {
+                let mut buffer = Vec::new();
+                move |input, output| {
+                    input.for_each(|cap, data| {
+                        data.swap(&mut buffer);
+
+                        let diff = buffer.iter().map(|(_d, _t, r)| r).sum();
+                        logger.log(ComputeEvent::ErrorCount { export_id, diff });
+
+                        output.session(&cap).give_vec(&mut buffer);
+                    });
+                }
+            })
+            .as_collection()
+    }
+}
+
+impl<G, B> LogDataflowErrors for Stream<G, B>
```

Comment:
Do we need this implementation? The arrangement that we're reading from already logs the counts, so we could re-use this without needing to count again. It sure makes the queries more complicated, but would not incur the cost of `sum_batch_diffs`.

Reply:
> The arrangement that we're reading from already logs the counts

You mean the arrangement size logging, right? That logs the number of physical records, not the logical number of errors. That count can be wildly different if the arrangement isn't fully compacted, so I don't think we can use it. See my comment above for an example that shows how misleading that would be.